demo/text-fields.html

```diff
@@ -104,32 +108,114 @@ <h3 demo-section>States</h3>
       </template>
     </nice-demo-snippet>
 
+    <h3 demo-section>Types</h3>
+    <nice-demo-snippet>
+      <template slot="source">
+        <vaadin-text-field label="Text" value="Value"></vaadin-text-field>
+        <vaadin-password-field label="Password" value="Secret –&nbsp;hush!" placeholder="Type a secret"></vaadin-password-field>
+        <br>
+        <vaadin-text-area label="Text area" value="Value" placeholder="Placeholder"></vaadin-text-area>
+      </template>
+    </nice-demo-snippet>
+
     <h3 demo-section>Dense layout</h3>
     <nice-demo-snippet>
       <template slot="source">
         <vaadin-text-field theme="dense" label="Label" required error-message="Error message"></vaadin-text-field>
       </template>
     </nice-demo-snippet>
 
-    <h3 demo-section>Dark theme</h3>
+    <h2 demo-section>Dark theme</h3>
+
+    <h3 demo-section>Label and placeholder</h3>
     <nice-demo-snippet>
       <template slot="source">
         <div theme="dark">
+          <vaadin-text-field placeholder="Placeholder"></vaadin-text-field>
+
+          <vaadin-text-field label="Label"></vaadin-text-field>
 
-          <vaadin-text-field label="Label" placeholder="Placeholder"></vaadin-text-field>
+          <vaadin-text-field label="Label and placeholder" placeholder="Placeholder"></vaadin-text-field>
 
+          <vaadin-text-field label="Label always floating" theme="always-float-label"></vaadin-text-field>
+        </div>
+      </template>
+    </nice-demo-snippet>
+
+    <h3 demo-section>Prefix and suffix</h3>
+    <nice-demo-snippet>
+      <template slot="source">
+        <div theme="dark">
+          <vaadin-text-field placeholder="Search">
+            <iron-icon slot="prefix" icon="vaadin:search"></iron-icon>
+          </vaadin-text-field>
+
+          <vaadin-text-field label="Length"
+                             theme="always-float-label"
+                             placeholder="123"
+                             pattern="[0-9]{0,3}"
+                             prevent-invalid-input
+                             style="width: 6ch;">
+            <span slot="suffix">cm</span>
+          </div>
+        </vaadin-text-field>
+
+        <custom-style>
+          <style>
+            iron-icon[slot] {
+              --iron-icon-width: 1em;
+              --iron-icon-height: 1em;
+              padding: 4px;
+              margin-right: 4px;
```

Comment:
Should we move these to the text field theme instead? At least in Valo, iron-icons are sized by default.